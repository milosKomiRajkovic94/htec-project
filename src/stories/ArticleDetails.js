import React from "react";

import PropTypes from "prop-types";

import "./articleDetails.css";

export const ArticleDetails = ({ singleArticleData, onBack }) => {
  return (
    <div className="singleArticleDetails">
      <div
        className={
          singleArticleData.title
            ? "singleArticleDetailsTitle"
            : "singleArticleDetailsTitle visibilityHidden"
        }
      >
        <li>{singleArticleData.title}</li>
      </div>
      <div className="singleArticleDetailsImage">
        <img
          src={
            singleArticleData.urlToImage
              ? singleArticleData.urlToImage
              : require("../assets/images/boxImage.png")
          }
          alt="singleArticleDetailsImageTag"
          className="singleArticleDetailsImageTag"
        />
      </div>
      <div
        className={
          singleArticleData.content
            ? "singleArticleDetailsContent"
            : "singleArticleDetailsContent visibilityHidden"
        }
      >
        {singleArticleData.content}
      </div>
      <div className="singleArticleDetailsContentBack">
        <span
          className="cursorPointer"
          onClick={() => onBack()}
          onKeyPress={() => onBack()}
          tabIndex={0}
          role="button"
        >
          {"< Back to list"}
        </span>
      </div>
    </div>
  );
};

ArticleDetails.propTypes = {
  singleArticleData: PropTypes.shape({
    title: PropTypes.string,
    urlToImage: PropTypes.string,
    content: PropTypes.string,
  }),
  onBack: PropTypes.func,
};

ArticleDetails.defaultProps = {
  singleArticleData: PropTypes.shape({
    title: "",
    urlToImage: "",
    content: "",
  }),
  onBack: () => {},
};
