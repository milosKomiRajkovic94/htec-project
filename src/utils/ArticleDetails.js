import React from "react";

import PropTypes from "prop-types";

import { useTranslation } from "react-i18next";

const ArticleDetails = ({ singleArticleData, onBack }) => {
  const { t } = useTranslation();

  return (
    <div className="singleArticleDetails">
      <div
        className={
          singleArticleData.title
            ? "singleArticleDetailsTitle"
            : "singleArticleDetailsTitle visibilityHidden"
        }
      >
        <li>{singleArticleData.title}</li>
      </div>
      <div className="singleArticleDetailsImage">
        <img
          src={
            singleArticleData.urlToImage
              ? singleArticleData.urlToImage
              : require("../assets/images/boxImage.png")
          }
          alt="singleArticleDetailsImageTag"
          className="singleArticleDetailsImageTag"
        />
      </div>
      <div
        className={
          singleArticleData.content
            ? "singleArticleDetailsContent"
            : "singleArticleDetailsContent visibilityHidden"
        }
      >
        {singleArticleData.content}
      </div>
      <div className="singleArticleDetailsContentBack">
        <span
          className="cursorPointer"
          onClick={() => onBack()}
          onKeyPress={() => onBack()}
          tabIndex={0}
          role="button"
          data-testid="back"
        >
          {`< ${t("backToList")}`}
        </span>
      </div>
    </div>
  );
};

ArticleDetails.propTypes = {
  singleArticleData: PropTypes.shape({
    title: PropTypes.string,
    urlToImage: PropTypes.string,
    content: PropTypes.string,
  }),
  onBack: PropTypes.func,
};

ArticleDetails.defaultProps = {
  singleArticleData: PropTypes.shape({
    title: "",
    urlToImage: "",
    content: "",
  }),
  onBack: () => {},
};

export default ArticleDetails;
