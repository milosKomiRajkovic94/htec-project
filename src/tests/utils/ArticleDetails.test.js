import React from "react";

import ReactDOM from "react-dom";

import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import ArticleDetails from "../../utils/ArticleDetails";

afterEach(cleanup);

it("renders without crashing with four required props", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <ArticleDetails
      singleArticleData={{
        title: "something",
        content: "something",
        url: "something",
      }}
    />,
    div
  );
});

it("renders language buttons correctly", () => {
  const { getByTestId } = render(
    <ArticleDetails
      singleArticleData={{
        title: "something",
        content: "something",
        url: "something",
      }}
    />
  );
  expect(getByTestId("back")).toHaveTextContent("backToList");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <ArticleDetails
        singleArticleData={{
          title: "something",
          content: "something",
          url: "something",
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
