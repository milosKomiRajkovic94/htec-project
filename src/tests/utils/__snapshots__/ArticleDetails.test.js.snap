// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="singleArticleDetails"
>
  <div
    className="singleArticleDetailsTitle"
  >
    <li>
      something
    </li>
  </div>
  <div
    className="singleArticleDetailsImage"
  >
    <img
      alt="singleArticleDetailsImageTag"
      className="singleArticleDetailsImageTag"
      src="boxImage.png"
    />
  </div>
  <div
    className="singleArticleDetailsContent"
  >
    something
  </div>
  <div
    className="singleArticleDetailsContentBack"
  >
    <span
      className="cursorPointer"
      data-testid="back"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      &lt; backToList
    </span>
  </div>
</div>
`;
